src/apple/deps/mod.rs

```diff
@@ -33,23 +39,153 @@ pub enum Error {
     PromptFailed(#[from] std::io::Error),
     #[error(transparent)]
     VersionLookupFailed(#[from] system_profile::Error),
+    #[error("Failed to update package `{package}`")]
+    PackageNotUpdated { package: &'static str },
+    #[error("Failed to list installed gems: {0}")]
+    GemListFailed(#[from] bossy::Error),
+    #[error("Regex match failed for output of `gem list`")]
+    RegexMatchFailed,
+    #[error(transparent)]
+    CaptureGroupError(#[from] util::CaptureGroupError),
+}
+
+#[derive(Default)]
+pub struct GemCache {
+    set: HashSet<String>,
 }
 
-pub fn install(package: &'static str, reinstall_deps: opts::ReinstallDeps) -> Result<bool, Error> {
-    let found = util::command_present(package)
-        .map_err(|source| Error::PresenceCheckFailed { package, source })?;
-    log::info!("package `{}` present: {}", package, found);
-    if !found || reinstall_deps.yes() {
-        println!("Installing `{}`...", package);
-        // reinstall works even if it's not installed yet, and will upgrade
-        // if it's already installed!
-        bossy::Command::impure_parse("brew reinstall")
+impl GemCache {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn initialize(&mut self) -> Result<(), Error> {
+        if self.set.is_empty() {
+            self.set = bossy::Command::impure_parse("gem list")
+                .run_and_wait_for_string()
+                .map_err(Error::GemListFailed)?
+                .lines()
+                .flat_map(|string| {
+                    regex!(r"(?P<name>.+) \(.+\)").captures(string).map(|caps| {
+                        util::get_string_for_group(&caps, "name", string)
+                            .map_err(Error::CaptureGroupError)
+                    })
+                })
+                .collect::<Result<_, Error>>()?;
+        }
+        Ok(())
+    }
+
+    pub fn contains(&mut self, package: &str) -> Result<bool, Error> {
+        self.initialize()?;
+        Ok(self.contains_unchecked(package))
+    }
+
+    pub fn contains_unchecked(&self, package: &str) -> bool {
+        self.set.contains(package)
+    }
+
+    pub fn reinstall(&mut self, package: &'static str) -> Result<(), Error> {
+        let command = if self.contains(package)? {
+            "gem update"
+        } else {
+            println!("`sudo` is required to install {:?} using gem", package);
```

Comment:
```suggestion
            println!("`sudo` is required to install {} using gem", package);
```
After seeing the output in person, this makes it a bit more natural